packages/ui/src/components/pressable/pressable.native.tsx

```diff
@@ -11,44 +16,74 @@ import {
   VisibleProps,
   backgroundColorShorthand,
   border,
-  createRestyleComponent,
+  composeRestyleFunctions,
   layout,
   opacity,
   position,
   spacingShorthand,
+  useRestyle,
   visible,
 } from '@shopify/restyle';
 
+import { isString } from '@leather.io/utils';
+
+import { useHaptics } from '../../hooks/use-haptics.native';
 import { Theme } from '../../theme-native';
 
-export type PressableBaseProps = RNPressableProps &
-  OpacityProps<Theme> &
+type PressableHapticFeedbackType = 'soft' | 'light' | 'medium' | 'heavy' | 'rigid';
+
+interface HapticConfig {
+  onPress?: PressableHapticFeedbackType;
+  onLongPress?: PressableHapticFeedbackType;
+}
+
+type RestyleProps = OpacityProps<Theme> &
   VisibleProps<Theme> &
   SpacingShorthandProps<Theme> &
   BorderProps<Theme> &
   BackgroundColorShorthandProps<Theme> &
   LayoutProps<Theme> &
   PositionProps<Theme>;
 
-export const buttonRestyleFunctions = [
+type PressableElement = ElementRef<typeof RNPressable>;
+
+interface PressableBaseProps extends RNPressableProps, RestyleProps {
+  haptics?: PressableHapticFeedbackType | HapticConfig;
```

Comment:
I'd say leaner API design is better, if I were to choose between removing the string annotation vs. having two props instead of one, I'd pick the former every time, but then like you said, `haptics="soft"` is just a convenience shortcut for the most commonly used use-case.
I'd say what this really lacks right now is proper JSDoc annotation + docs in Storybook.